src/Analyzers/CONTRIBUTING.md

```diff
@@ -0,0 +1,161 @@
+- [Roslyn analyzers and code-aware library for ImmutableArrays](https://learn.microsoft.com/en-us/visualstudio/extensibility/roslyn-analyzers-and-code-aware-library-for-immutablearrays?view=vs-2022)
+- [Roslyn code samples](https://github.com/dotnet/roslyn/blob/main/docs/analyzers/Analyzer%20Samples.md)
+
+## Unit Testing
+
+There is a [test project](../../test/Analyzers.Tests/) that contains several unit tests for Durable Task analyzers.
+We also have some verifier methods that allows you to quickly create the code snippets you would like to test,
+such as:
+
+- `CSharpAnalyzerVerifier<TAnalyzer>.VerifyDurableTaskAnalyzerAsync(string source, params DiagnosticResult[] expected)`
+- `CSharpCodeFixVerifier<TAnalyzer, TCodeFix>.VerifyDurableTaskAnalyzerAsync(string source, params DiagnosticResult[] expected)`
+- `CSharpCodeFixVerifier<TAnalyzer, TCodeFix>.VerifyDurableTaskAnalyzerAsync(string source, params DiagnosticResult[] expected, string fix)`
+
+For instance, if you would like to test `MatchingInputOutputTypeActivityAnalyzer`.
+You can start creating a test class and add the following type alias:
+
+```cs
+using VerifyCS = Microsoft.DurableTask.Analyzers.Tests.Verifiers.CSharpAnalyzerVerifier<Microsoft.DurableTask.Analyzers.Activities.MatchingInputOutputTypeActivityAnalyzer>;
+```
+
+Then, you can use the verifier in your test code:
+
+```cs
+[Fact]
+public async Task DurableFunctionActivityInvocationWithMismatchedInputType()
+{
+    string code = @"
+<the cs code containing a violation>
+";
+
+    // build the expected diagnostic
+    DiagnosticResult expected = VerifyCS
+                                    .Diagnostic(MatchingInputOutputTypeActivityAnalyzer.InputArgumentTypeMismatchDiagnosticId)
+                                    .WithArguments("something");
+
+    // asserts that the diagnostic is reported
+    await VerifyCS.VerifyDurableTaskAnalyzerAsync(code, expected);
+}
+```
+
+For additional details on how to test an analyzer, please see [this tutorial](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix#build-unit-tests).
+
+## Manual Testing
+
+Any of the [samples projects](../../samples/) in this repository can be used for manual testing and validation.
+
+You can just add a project reference to `..\..\src\Analyzers\Analyzers.csproj`:
+
+```xml
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\Analyzers\Analyzers.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+  </ItemGroup>
+```
+
+> When using Visual Studio, you may need to restart the IDE, since analyzers and code fixers are usually loaded and cached during start up.
+Be aware of this behavior when developing new features as well.
+
+### Debugging
+
+Although debugging the unit tests are a great way to inspect an analyzer internal state,
+it can be useful to attach a debugger to one the samples projects mentioned above.
```

Comment:
```suggestion
Although debugging the unit tests is a great way to inspect an analyzer's internal state,
it can be useful to attach a debugger to one of the samples projects mentioned above.
```